### PR TITLE
Use system TBB by default for Deb, RPM

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
### Details:
 - Otherwise, currently users have error during Debian package build

### CODE REVIEW

File: cmake/features.cmake [hunk 1]
```diff
@@ -162,7 +162,15 @@ else()
     set(ENABLE_SYSTEM_FLATBUFFERS_DEFAULT ON)
 endif()
 
-ov_dependent_option (ENABLE_SYSTEM_TBB  "Enables use of system TBB" OFF "THREADING MATCHES TBB" OFF)
+# use system TBB only for Debian / RPM packages
+if(CPACK_GENERATOR MATCHES "(DEB|RPM)")
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:166]:
```suggestion
if(CPACK_GENERATOR MATCHES "^(DEB|RPM)$")
```

File: cmake/features.cmake [hunk 2]
```diff
@@ -162,7 +162,15 @@ else()
     set(ENABLE_SYSTEM_FLATBUFFERS_DEFAULT ON)
 endif()
 
-ov_dependent_option (ENABLE_SYSTEM_TBB  "Enables use of system TBB" OFF "THREADING MATCHES TBB" OFF)
+# use system TBB only for Debian / RPM packages
+if(CPACK_GENERATOR MATCHES "^(DEB|RPM)$")
```
Comment (xufang-lisa) [inline cmake/features.cmake:166]: Is this modification due to (https://github.com/openvinotoolkit/openvino/blob/4b0868c98a32d888fe240d86669101ff32e07681/src/cmake/install_tbb.cmake#L98)?
Do other cpack types (like `CONDA-FORGE|BREW|CONAN|VCPKG`) need to be added here?

Comment (ilya-lavrenov) [inline cmake/features.cmake:166]:
```suggestion
if(CPACK_GENERATOR MATCHES "^(DEB|RPM|CONDA-FORGE|BREW|CONAN|VCPKG)$")
```

Comment (ilya-lavrenov) [reply cmake/features.cmake]: good catch, thank you!

**Review by peterchen-intel — APPROVED** [review]

**Review by xufang-lisa — APPROVED** [review]

**Pull request merged**